using_restricted_array.rb

```diff
@@ -7,41 +8,89 @@
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
 def length(array)
-  puts "NOT IMPLEMENTED"
+  index = 0
+  while array[index] != nil
+    index += 1
+  end
+  return index
 end
 
 # Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+  index = 0
+  while index < length(array)
+    print "#{array[index]} "
+    index += 1
+  end
+  print "\n"
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
 def search(array, length, value_to_find)
-  puts "NOT IMPLEMENTED"
+  index = 0
+  while index < length
+    return array[index] if value_to_find == array[index]
```

Comment:
Pedantic; since the specifications ask to return true or false, please return it explicitly instead of relying on Ruby's truthy and falsy properties. To that effect, make line 33 return true and return false after line 35 i.e. outside the loop.